nextcord/cache.py

```diff
@@ -0,0 +1,149 @@
+# SPDX-License-Identifier: MIT
+
+from enum import Enum
+from time import time as now
+from typing import Hashable, Optional, Union
+
+# Avoiding super() call speeds up by 2x
+# Cache speedup is very beneficial as it will be one of
+# the hotest spots called
+__set_item__ = dict.__setitem__
+__get_item__ = dict.__getitem__
+__get__ = dict.get
+__pop__ = dict.pop
+
+
+class CacheMissing(Enum):
+    MISSING = 0
+
+
+_MISSING = CacheMissing.MISSING

```

Comment:
Why not `MISSING` from `nextcord.utils`?

Reply:
> Why not `MISSING` from `nextcord.utils`?
Because in this case the cache is expected to be able to store nextcord's MISSING object, without its own MISSING, it could cause issues with ttl & returning content